x-pack/plugins/observability/public/pages/slos/components/common/sort_by_select.tsx

```diff
@@ -5,64 +5,132 @@
-      compressed
-      prepend={SORT_BY_LABEL}
-      options={SORT_OPTIONS}
-      value={initialState.sort.by}
-      onChange={(evt) => {
-        onStateChange({
-          page: 0,
-          sort: { by: evt.target.value as SortField, direction: initialState.sort.direction },
+export type Item<T> = EuiSelectableOption & {
+  label: string;
+  type: T;
+  checked?: EuiSelectableOptionCheckedType;
+};
+
+export function SLOSortBy({ initialState, onStateChange, sortBy }: Props) {
+  const [isSortByPopoverOpen, setIsSortByPopoverOpen] = useState(false);
+  const handleChangeSortBy = ({ value, label }: { value: SortField; label: string }) => {
+    setGroupLabel(label);
+    onStateChange({
+      page: 0,
+      sort: { by: value, direction: initialState.sort.direction },
+    });
+  };
+
+  const sortByOptions: Option[] = [
+    {
+      label: i18n.translate('xpack.observability.slo.list.sortBy.sliValue', {
+        defaultMessage: 'SLI value',
+      }),
+      checked: sortBy === 'sli_value',
+      value: 'sli_value',
+      onClick: () => {
+        handleChangeSortBy({
+          value: 'sli_value',
+          label: i18n.translate('xpack.observability.slo.list.sortBy.sliValue', {
+            defaultMessage: 'SLI value',
+          }),
+        });
+      },
+    },
+    {
+      label: i18n.translate('xpack.observability.slo.list.sortBy.sloStatus', {
+        defaultMessage: 'SLO status',
+      }),
+      checked: sortBy === 'status',
+      value: 'status',
+      onClick: () => {
+        handleChangeSortBy({
+          value: 'status',
+          label: i18n.translate('xpack.observability.slo.list.sortBy.sloStatus', {
+            defaultMessage: 'SLO status',
+          }),
+        });
+      },
+    },
+    {
+      label: i18n.translate('xpack.observability.slo.list.sortBy.errorBudgetConsumed', {
+        defaultMessage: 'Error budget consumed',
+      }),
+      checked: sortBy === 'error_budget_consumed',
+      value: 'error_budget_consumed',
+      onClick: () => {
+        handleChangeSortBy({
+          value: 'error_budget_consumed',
+          label: i18n.translate('xpack.observability.slo.list.sortBy.errorBudgetConsumed', {
+            defaultMessage: 'Error budget consumed',
+          }),
+        });
+      },
+    },
+    {
+      label: i18n.translate('xpack.observability.slo.list.sortBy.errorBudgetRemaining', {
+        defaultMessage: 'Error budget remaining',
+      }),
+      checked: sortBy === 'error_budget_remaining',
+      value: 'error_budget_remaining',
+      onClick: () => {
+        handleChangeSortBy({
+          value: 'error_budget_remaining',
+          label: i18n.translate('xpack.observability.slo.list.sortBy.errorBudgetRemaining', {
+            defaultMessage: 'Error budget remaining',
+          }),
         });
-      }}
+      },
+    },
+  ];
+
+  const [groupLabel, setGroupLabel] = useState(
+    sortByOptions.find((option) => option.value === sortBy)?.label || 'Default'
```

Comment:
Should we default to the first `sortByOptions[0].label` in case sortBy does not match any existing values?